### PR TITLE
Flags identified as matched incorrectly 

### PR DESCRIPTION
**Pull request by 5cr0ll**
During running of the CIS checks, there are checks which give incorrect results, that is FAIL when
expected to be PASS and vice-versa. The reason is that `execute()` function does string parsing
incorrectly which results in false positive/negative results. Specifically the `strings.Contains()` is used which can mistakenly find prefix of the matched flag and not the flag itself. Another reason is that there are cases when `execute()` just checks that flag is not present when it is expected to do
more careful checking. For example: check 2.5. The check returns FAIL when actually docker does not have aufs storage driver set.

Fixed by changing execute() flow using correct regex to fetch the flag value.

### CODE REVIEW

**Comment by lizrice** [comment]
I'm not sure this is fixing the issue? [Try this on Go playground](https://play.golang.org/p/KG-3J9mVnAP) where I've written some tests (I think it would be a good idea to add some tests into the code for this while fixing it).

See also https://github.com/aquasecurity/kube-bench/issues/137 which is related, and which @ttousai was working on. The regex he has been testing is this: 
```
pttn := `(` + flag + `)([=\s])+([^\s]*) *`
```

That regex solves most of the cases, except for where we are looking for `--flag` with no value. I had suggested that in this there is no flagVal to find, we just need to know if it's present or not. If that means that the operation is always “has” or “nothave”, maybe we can use a different regex when that is the operation? @ttousai did you have a chance to try that approach out?

**Comment by semion-a** [comment]
@lizrice , there was error in regex, but the 2.15 check still worked correctly, thank you for catching that!
The error is that in test1 the --flag is matched inside the "--flag-should-not-match-this=/etc/foobar" string. The match is done by `(`+flag+`)` (3'rd pattern). 
IMO Can be fixed by  `(`+flag +`\s|$)` instead. The '$' is for the case when --flag is at the end of cmd line. This fix causes `FindStringSubmatch()` to return always a nonempty slice and therefore I changed `matchExpected` to true in https://play.golang.org/p/Sf-DcTqO91K

Regarding the case when --flag with no value is used:
The regex covers this case. The `FindStringSubmatch()` returns --flag and I am not sure I understand why "" put in the `match` field of the test3.

In case we're sure that we don't want --flag to be matched by regex, but be handled by "has", then the `(` + flag + `)([=\s])+([^\s]*) *` can be used instead.

Please let me know what changes are needed.

**Comment by lizrice** [comment]
I think I've got a regex that fixes all the cases!  
```
	pttn := `(` + flag + `)([=\s])+([^\s-]*) *`
```

- Matches on the flag, which must be followed by either = or whitespace. 
- Extracted value can't begin with - (as that would be another flag). 

Check out the tests here: https://play.golang.org/p/PmRolYFuqxZ. Note that I had to trim spaces (to avoid a match on whitespace) but no flag value can begin or end with whitespace anyway. 

> Regarding the case when --flag with no value is used:
> The regex covers this case. The FindStringSubmatch() returns --flag and I am not sure I understand why "" put in the match field of the test3.

If --flag has no value, surely we want the match string to be empty? We should be able to use the fact that there is any match at all (as in the tests). 

@ttousai wdyt? Can you try this for kube-bench 137?

**Comment by semion-a** [comment]
@lizrice , in project's master branch: in `test.go` in `execute()` function:
```			
case "has":
   result = strings.Contains(flagVal, t.Compare.Value)
```

The flagVal is read from vals returned by `FindStringSubmatch()`:
```
....
flagVal = vals[1]
....
```
This means that for e.g. check 1.1 the /var/lib/docker is in `vals[1]` and `strings.Contains()` checks that `/var/lib/docker` contains `/var/lib/docker.`

In the PR the logic is similar:
```
for i, currentValue := range vals {
					if i == 0 {
						continue
					}
					if len(currentValue) > 0 {
						flagVal = currentValue
					}
}
```
Which means go over the `vals` (not including 0'th member) and find matched value.
For 1.1. the value will be in `vals[3]` as the (` + flag + `(?: +|$))` is matched. See https://play.golang.org/p/cXt37Fd2PmA

>causes FindStringSubmatch() to return always a nonempty slice

After using non-capturing group `(?: +|$)` the vals would be empty, changed `matchExpected` to false as it was.

Note: the `(` + flag + `)([=\s])+([^\s-]*) *` pattern does not catch this case:  https://play.golang.org/p/GvTMGDWz9W0

**Comment by lizrice** [comment]
@semion-a I think we're agreed that the we've got a better regex, now but that the logic that follows in case of no match is wrong, and shouldn't os.Exit(). You're planning to work on that, so I'll assign this PR to you now  (I'm also going to change the title of this PR to describe the issue better)

**Comment by lizrice** [comment]
@semion-a well, I would assign you if it would let me! I invited you as a collaborator

**Comment by semion-a** [comment]
Accepted invitation.

**Comment by semion-a** [comment]
@lizrice , updated PR. Please do not merge it until PR for docker-bench is merged, as it changes the checks in definitions.yaml.

**Review by lizrice — CHANGES_REQUESTED** [review]
As I said before, I want to see some tests in the code for this - it's a lot to ask a reviewer to approve without tests. You can easily take the work I already did in the Go Playground earlier and convert that into Go tests that would prove whether or not this is working.

**Review by lizrice — APPROVED** [review]
LGTM. Thanks for the tests!

**Pull request merged**
